test/runtest.py

```diff
@@ -1543,6 +1543,50 @@ def f():
       time.sleep(0.1)
 
     self.assertEqual(found_message, True)
+
+    ray.worker.cleanup()
+
+  def testTaskProfileAPI(self):
+    ray.init(redirect_output=True)
+
+    @ray.remote
+    def f():
+      return 1
+
+    num_calls = 5
+    [f.remote() for _ in range(num_calls)]
+
+    # Make sure the event log has the correct number of events.
+    start_time = time.time()
+    while time.time() - start_time < 10:
+      profiles, events = ray.global_state.task_profiles()
+      if len(profiles) == num_calls:
+        break
+      time.sleep(0.1)
+    self.assertEqual(len(profiles), num_calls)
+    self.assertEqual(len(events), 0)
```

Comment:
I made a few small changes because I don't think we were waiting to make sure enough elements appeared in `profiles`.